thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/DetectionHealth.java

```diff
@@ -0,0 +1,379 @@
+
+      // construct regression status
+      RegressionStatus status = new RegressionStatus();
+      status.detectorMapes = detectorMapes;
+      status.detectorHealthStatus = detectorMapes.entrySet()
+          .stream()
+          .collect(Collectors.toMap(Map.Entry::getKey, e -> classifyMapeHealth(e.getValue())));
+      status.healthStatus = classifyOverallRegressionStatus(status.detectorHealthStatus);
+      return status;
+    }
+
+    private static HealthStatus classifyMapeHealth(double mape) {
+      if (mape < 0.2) {
+        return HealthStatus.GOOD;
+      }
+      if (mape < 0.5) {
+        return HealthStatus.MODERATE;
+      }
+      return HealthStatus.BAD;
+    }
+
+    /**
+     * Classify the regression status of the detection config based on the health status for each detector
+     * @param detectorHealthStatus the health status for each detector
+     * @return the overall regression status
+     */
+    private static HealthStatus classifyOverallRegressionStatus(Map<String, HealthStatus> detectorHealthStatus) {
+      if (detectorHealthStatus.values().contains(HealthStatus.GOOD)) {
+        return HealthStatus.GOOD;
+      }
+      if (detectorHealthStatus.values().contains(HealthStatus.MODERATE)) {
+        return HealthStatus.MODERATE;
+      }
+      return HealthStatus.BAD;
+    }
+
+    private AnomalyCoverageStatus buildAnomalyCoverageStatus() {
+      List<MergedAnomalyResultDTO> anomalies = this.anomalyDAO.findByPredicate(
+          Predicate.AND(Predicate.LT(COL_NAME_START_TIME, this.endTime),
+              Predicate.GT(COL_NAME_END_TIME, this.startTime),
+              Predicate.EQ(COL_NAME_DETECTION_CONFIG_ID, detectionConfigId)));
+      anomalies = anomalies.stream().filter(anomaly -> !anomaly.isChild()).collect(Collectors.toList());
+
+      // the anomalies can come from different sub-dimensions, merge the anomaly range if possible
+      List<Interval> intervals = new ArrayList<>();
+      if (!anomalies.isEmpty()) {
+        anomalies.sort(Comparator.comparingLong(MergedAnomalyResultBean::getStartTime));
+        long start = anomalies.stream().findFirst().get().getStartTime();
+        long end = anomalies.stream().findFirst().get().getEndTime();
+        for (MergedAnomalyResultDTO anomaly : anomalies) {
+          if (anomaly.getStartTime() <= end) {
+            end = Math.max(end, anomaly.getEndTime());
+          } else {
+            intervals.add(new Interval(start, end));
+            start = anomaly.getStartTime();
+            end = anomaly.getEndTime();
+          }
+        }
+        intervals.add(new Interval(start, end));
+      }
+      long totalAnomalyCoverage =
+          intervals.stream().map(interval -> interval.getEndMillis() - interval.getStartMillis()).reduce(0L, Long::sum);
+
+      AnomalyCoverageStatus coverageStatus = new AnomalyCoverageStatus();
+      coverageStatus.anomalyCoverageRatio = (double) totalAnomalyCoverage / (this.endTime - this.startTime);
```

Comment:
Do we perform any validation on endTime and startTime? What if start is greater than end or if both are same!

Reply:
We have this validation in the constructor